### PR TITLE
Fixes #15148: Add search/pagination to eventlogs UI

### PR DESCRIPTION
**Pull request by ElaadF**
https://issues.rudder.io/issues/15148

### CODE REVIEW

**Comment by ElaadF** [comment]
PR rebased

File: webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/web/services/RudderProperties.scala [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+import com.normation.rudder.domain.logger.ApplicationLogger
+import com.typesafe.config.{Config, ConfigFactory}
+import org.springframework.core.io.ClassPathResource
+/**
+ * Define a resource for configuration.
+ * For now, config properties can only be loaded from either
+ * a file in the classpath, or a file in the file system.
+ */
+sealed trait ConfigResource
```
Comment (VinceMacBuche) [inline webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/web/services/RudderProperties.scala]: I don'tthink this should be here

File: webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/web/services/DiffDisplayer.scala [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+import com.normation.rudder.domain.policies.DirectiveId
+import com.normation.rudder.repository.{FullNodeGroupCategory, RoRuleRepository}
+import com.normation.rudder.rule.category.{RuleCategory, RuleCategoryId, RuleCategoryService}
+import net.liftweb.common.Full
+import net.liftweb.common.EmptyBox
+import net.liftweb.common.Loggable
+import com.normation.rudder.domain.policies._
+import com.normation.rudder.repository.ldap.{LDAPEntityMapper, RoLDAPNodeGroupRepository, RoLDAPRuleRepository, ScalaLock}
+import com.normation.rudder.web.model.LinkUtil
+import RudderProperties._
```
Comment (VinceMacBuche) [inline webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/web/services/DiffDisplayer.scala]: I don't think you should use properties here, what is used in RudderProperties ?

File: webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/web/services/RudderProperties.scala (hunk 1 above)
Comment (VinceMacBuche) [reply webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/web/services/RudderProperties.scala]: I think you don't need that file

File: webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/web/services/DiffDisplayer.scala [hunk 3]
```diff
@@ -0,0 +1,257 @@
+/*
```
Comment (VinceMacBuche) [inline webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/web/services/DiffDisplayer.scala]: That code should be removed from Rudder-web

File: webapp/sources/rudder/rudder-rest/pom.xml [hunk 4]
```diff
@@ -98,6 +98,11 @@ along with Rudder.  If not, see <http://www.gnu.org/licenses/>.
   </properties>
 
   <dependencies>
+    <dependency>
+      <groupId>org.springframework.security</groupId>
```
Comment (VinceMacBuche) [inline webapp/sources/rudder/rudder-rest/pom.xml]: I think this is used By RUdderProperties but i do feel we shouldn't have RudderProperties in Rudder-rest only in Rudder-web

File: webapp/sources/rudder/rudder-web/src/main/scala/com/normation/rudder/web/services/EventListDisplayer.scala [hunk 5]
```diff
@@ -80,6 +80,7 @@ import com.normation.rudder.rule.category.RoRuleCategoryRepository
 import org.joda.time.format.DateTimeFormat
 import com.normation.rudder.web.model.LinkUtil
 import org.joda.time.format.ISODateTimeFormat
+import scala.concurrent.Future
 import com.normation.box._
```
Comment (VinceMacBuche) [inline webapp/sources/rudder/rudder-web/src/main/scala/com/normation/rudder/web/services/EventListDisplayer.scala]: I think you need to delete the code that is not used anymore to display a events here

**Comment by ElaadF** [comment]
Commit modified

**Comment by ElaadF** [comment]
Commit modified

File: webapp/sources/rudder/rudder-rest/pom.xml (hunk 4 above)
Comment (VinceMacBuche) [reply webapp/sources/rudder/rudder-rest/pom.xml]: @ElaadF is it compiling without that dependency ?

Comment (VinceMacBuche) [reply webapp/sources/rudder/rudder-rest/pom.xml]: ok it's a dependency used in CurrentUserService

Comment (VinceMacBuche) [reply webapp/sources/rudder/rudder-rest/pom.xml]: It should be remvoed from rudder-web/pom.xml (if no compilation problem)

File: webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/web/services/EventLogDetailsService.scala [hunk 6]
```diff
@@ -0,0 +1,1567 @@
+/*
+*************************************************************************************
+* Copyright 2011 Normation SAS
```
Comment (VinceMacBuche) [inline webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/web/services/EventLogDetailsService.scala]: Can you change the year to 2019 ?

File: webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/web/services/RudderUserDetails.scala [hunk 7]
```diff
@@ -0,0 +1,64 @@
+package com.normation.rudder.web.services
```
Comment (VinceMacBuche) [inline webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/web/services/RudderUserDetails.scala]: Can You add the License ?

File: webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/web/services/JsTableData.scala [hunk 8]
```diff
@@ -0,0 +1,71 @@
+/*
```
Comment (VinceMacBuche) [inline webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/web/services/JsTableData.scala]: Have you deleted the original file in rudder-web ? it should have been a git mv command ! just rm JsTableData.scala file in rudder-web it should link it

**Comment by ElaadF** [comment]
Commit modified

**Comment by ElaadF** [comment]
PR rebased

**Comment by ElaadF** [comment]
Commit modified

File: webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/internal/EventLogAPI.scala [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+      repos.getEventLogByCriteria(Some(s"id = $id")).toBox match {
+        case Full(e) =>
+          e.headOption match {
+            case Some(eventLog) =>
+              val crid = eventLog.id.flatMap(repos.getEventLogWithChangeRequest(_).toBox match {
+                case Full(Some((_, crId))) => crId
+                case _                     => None
+              })
+              val htmlDetails = eventLogDetail.displayDetails(eventLog, crid)
+              toJsonResponse(None, "content" -> htmlDetails.toString())("eventdetails", prettify = false)
```
Comment (fanf) [inline webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/internal/EventLogAPI.scala:151]: The toString here breaks JS script that used to be returned along with the HTML to diplay diff

**Pull request merged**